### PR TITLE
docs: add Open Graph metadata to support rich link preview

### PR DESCRIPTION
**Pull request by josedialpad**
## Description
Added [Open Graph](https://ogp.me/) metadata to the pages through [this plugin](https://vuepress-theme-hope.github.io/v2/seo/guide.html) so that social sites can unfold URLs based on this data.
- Renamed `desc` frontmatter prop to description that way [Vuepress](https://v2.vuepress.vuejs.org/reference/frontmatter.html#description) uses this to the meta description.
- For the component pages, used the thumbnail image for the `og:image` prop.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -125,5 +109,12 @@ export default defineUserConfig({
     googleAnalyticsPlugin({
       id: 'G-0YV8QJ44LF',
     }),
+    seoPlugin({
+      hostname: siteURL,
+      ogp: (ogp, page) => ({
+        ...ogp,
+        'og:image': siteURL + baseURL + page.frontmatter.image || ogp['og:image'],
```
Comment (josedialpad) [inline docs/.vuepress/config.js:116]: this seems to be the [default value](https://vuepress-theme-hope.github.io/v2/seo/guide.html#default-ogp-generation) (`frontmatter.image`) in the `og:image` prop when it's specified, though I couldn't get it working without explicitly adding it.

**Comment by josedialpad** [comment]
Example in Slack:

app slack com_client_T0YS8AK7X_D01UW8652HM

File: docs/.vuepress/config.js (hunk 1 above)
Comment (braddialpad) [reply docs/.vuepress/config.js:116]: hmm weird.. maybe a bug?

Comment (josedialpad) [reply docs/.vuepress/config.js:116]: Yeah maybe, I couldn't figure out in their codebase though

**Review by braddialpad — APPROVED** [review]
Stoked for this!

**Pull request merged**